src/Api/DataProvider/CoverItemDataProvider.php

```diff
@@ -34,12 +35,33 @@ public function getItem(string $resourceClass, $id, string $operationName = null
         $type = $this->getIdentifierType($context['request_uri']);
 
         $query = $this->buildElasticQuery($type, [$id]);
-        $searchResponse = $this->index->request('_search', Request::POST, $query->toArray());
-        $results = $searchResponse->getData();
+
+        // Note that we here don't uses the elastica request function to post the request to elasticsearch has we have
+        // had strange performance issues with it. We get information about the index and create the curl call by hand.
+        // We can do this as we now the complete setup and what should be taken into consideration.
+        $index = $this->index->getIndex();
```

Comment:
If this is repeat code from CoreCollectionDataProvider, please move to service or AbstractElasticSearchDataProvider

Reply:
Move to service since it is also used in 
src/Service/MoreInfoService/AbstractMoreInfoService.php

Reply:
Will move the first one, but not the last one as SOAP has to die.

Reply:
Fixed

Reply:
Ok